storage/range.go

```diff
@@ -1035,6 +1061,76 @@ func (r *Range) handleSkippedIntents(args proto.Request, intents []proto.Intent)
 	}
 }
 
+type chainedError struct {
+	error
+	cause *chainedError
+}
+
+// Error implements the error interface, printing the underlying chain of errors.
+func (ce *chainedError) Error() string {
+	if ce == nil {
+		ce = &chainedError{}
+	}
+	if ce.cause != nil {
+		return fmt.Sprintf("%s (caused by %s)", ce.error, ce.cause)
+	}
+	return ce.error.Error()
+}
+
+// newChainedError returns a chainedError made up from the given errors,
+// omitting nil values. It returns nil unless at least one of its arguments
+// is not nil.
+func newChainedError(errs ...error) *chainedError {
+	if len(errs) == 0 || (len(errs) == 1 && errs[0] == nil) {
+		return nil
+	}
+	ce := &chainedError{error: errs[0]}
+	for _, err := range errs[1:] {
+		if err == nil {
+			continue
+		}
+		ce.cause = &chainedError{error: err}
+		ce = ce.cause
+	}
+	return ce
+}
+
+// A replicaCorruptionError indicates that the replica has experienced an error
+// which puts its integrity at risk.
+type replicaCorruptionError struct {
+	error
+	processed bool // for now, required for testing
```

Comment:
can you expand this comment to indicate that "processed" means this error has been "handled"?